uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/modal/modal-document.js

```diff
@@ -119,6 +120,34 @@ define([
         AssignmentService.updateCache(obj);
       };
 
+      /**
+       * Searches the list of assignments for the given target contact
+       *
+       * @param  {string | integer} targetContactId
+       */
+      $scope.cacheTargetContactAssignments = function (targetContactId) {
+        AssignmentService.search(null, null).then(function (assignments) {
```

Comment:
Why get all assignments and then filter by contactId, add the contact id filter directly to the API call.